tensilelite/Tensile/KernelWriter.py

```diff
@@ -2680,12 +2680,14 @@ def initKernel(self, kernel, tensorParametersA, tensorParametersB):
     if kernel["ProblemType"]["Sparse"] and not kernel["DirectToVgprSparseMetadata"]:
       vwm = kernel["GlobalReadVectorWidthMetadata"]
 
-    if not kernel["UnrollMajorLDSA"]:
+    # use lrwvTile = 1 for XFP32
+    isXFP32 = kernel["EnableF32XdlMathOp"] and kernel["ProblemType"]["F32XdlMathOp"].isXFloat32()
+    if not kernel["UnrollMajorLDSA"] and not isXFP32:
```

Comment:
Could you simply check if data type is smaller than 4 bytes?  The issue is not only for XF32, but for datatype >= 4 bytes.

if not kernel["UnrollMajorLDSA"] and kernel["ProblemType"]["DataType"].numRegisters() < 1:
      self.states.lrvwTileA = kernel["VectorWidthA"]
 else:
      self.states.lrvwTileA = 1

Reply:
This issue is XFP32 specific.
It is because MatrixInstK is doubled for XFP32.
S,D,C,Z GEMM do not have same issue.

Reply:
As I explained in the above comment,
====
(2) VectorWidthA=2 case (incorrect result)
vgprVALUA0:1 = ds_read_b64 (for K=0-3)
vgprVALUA2:3 = ds_read_b64 (for K=4-7)
v_mfma vgprVALUA0:1,.... <-incorrect, we should combine VALUA0 and 2
====

This happens only with XFP32 because MatrixInstK is 8.

Reply:
Please add # FIXME for easy search.

Reply:
> This happens only with XFP32 because MatrixInstK is 8 (MI16x16x8x1).

This is common code. You shouldn't treat as "MatrixInstK is 8" for all platform.
The current un-supported condition for not UnrollMajorLDS is : data type >= 4 byte and MIInputPerThread is > 1

Reply:
OK. I will make another change as you suggest.

Reply:
I pushed the code change as you suggested.
I confirmed xfp32 test (I added in this PR) passed with the new change.